### PR TITLE
Adds Feature Flags documentation

### PR DESCRIPTION
**Pull request by afiune**
## Description
Adding documentation for the new Feature Flag work! Users need to know how
to turn features on and off.

tenor-178151550

~We shouldn't merge this until we have fixed an issue that @marcparadise found
where adding new fields to the `config.toml` will cause the `chef-run` command
to fail with:~
```
Invalid configuration entry.

When loading '$HOME/.chef-workstation/config.toml',
the configuration entry 'features' was not recognized.

Please check the spelling and ensure that the value is correct.
```

## To-Do
~If https://github.com/chef/chef-workstation-app/pull/125 is merged~

## All clear!
Ready for the @chef/docs-team to review this PR!

## Related Issue
https://github.com/chef/chef-workstation/issues/609

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: www/site/content/docs/reference/config.md [hunk 1]
```diff
@@ -379,3 +379,27 @@ Default
...
+#### Example
+
+```toml
+[features]
+preferences_dialog = true
+analyze = true
+```
+
+Description
+: List of experimental features that can be turn on and off. In the above example, there are two turned on features, the `preferences_dialog` feature, and the `analyze` feature. Every feature flag has a way to be turned on through the command line via environment variables. The respective environment variable for the `analize` feature would be `CHEF_FEAT_ANALYZE`. When set to any value, the feature will be turned on.
```
Comment (tyler-ball) [inline www/site/content/docs/reference/config.md]:
```suggestion
: List of experimental features that can be turn on and off. In the above example, there are two turned on features, the `preferences_dialog` feature, and the `analyze` feature. Every feature flag has a way to be turned on through environment variables. The respective environment variable for the `analyze` feature would be `CHEF_FEAT_ANALYZE`. When set to any value, the feature will be turned on.
```

File: www/site/content/docs/reference/config.md [hunk 2]
```diff
@@ -379,3 +379,27 @@ Default
...
+
+### Features
+
+Enable and disable experimental features for Chef Workstation.
+
+#### Example
+
+```toml
+[features]
+preferences_dialog = true
```
Comment (susanev) [inline www/site/content/docs/reference/config.md]: should this one be added later?

Comment (afiune) [reply www/site/content/docs/reference/config.md]: I used it as an example, I guess I can ask @kagarmoe to help me with another more generic example? 🤔  (I'm ok removing it or updating it)

Comment (kagarmoe) [reply www/site/content/docs/reference/config.md]: How does a user discover the possible features?

**Comment by afiune** [comment]
tenor-4289933

File: www/site/content/docs/reference/config.md (hunk 2 above)
Comment (afiune) [reply www/site/content/docs/reference/config.md]: @kagarmoe that is a great question! 💯 I have a proposal that is still under review here: 
https://github.com/chef/chef-workstation-app/pull/125

(it is mentioned in the description of this PR)

Product, UX, and Engineering are still working on a final solution but the idea is to have the
available features listed through the Chef Workstation App, the current design (that I made and won't be the final one) is:
Screen Shot 2019-11-08 at 1 40 26 AM

I would like to add that part somewhere in the docs, though, that would probably happen
in approx. two weeks (next week I'm on PTO), for now, this content is what we need until
we have the UI ready.

File: www/site/content/docs/reference/config.md [hunk 3]
```diff
@@ -299,83 +256,111 @@ Default
...
+analyze = true
+```
 
 Description
-: Set the update channel to use when checking for Chef Workstation updates. `"stable"` is the recommended value. Switch to `"current"` is not guaranteed to be stable, and should only be used if you are comfortable with the risks associated.
+: List of experimental features that can be turn on and off. In the above example, there are two turned on features, the `preferences_dialog` feature, and the `analyze` feature. Every feature flag has a way to be turned on through environment variables. The respective environment variable for the `analyze` feature would be `CHEF_FEAT_ANALYZE`. When set to any value, the feature will be turned on.
```
Comment (kagarmoe) [inline www/site/content/docs/reference/config.md]: The last sentence is confusing. Would this turn on both features?

```
[features]
preferences_dialog = true
analyze = false
```

Comment (afiune) [reply www/site/content/docs/reference/config.md]: Yes. I guess the confusion is that I probably should have added both environment variable names and not only one? 🤔 

> Every feature flag has a way to be turned on through environment variables. The corresponding environment variable for the `analyze` feature would be `CHEF_FEAT_ANALYZE`, and for the `preferences_dialog` would be `CHEF_FEAT_PREFERENCES_DIALOG`. When set to any value, the feature will be turned on.

Comment (afiune) [reply www/site/content/docs/reference/config.md]: No. In that snipped a user would have turned only one feature on.

File: www/site/content/docs/reference/config.md [hunk 4]
```diff
@@ -299,83 +256,111 @@ Default
...
-#### channel
+### Features
+
+Enable and disable experimental features for Chef Workstation.
+
+#### Example
+
+```toml
+[features]
+preferences_dialog = true
```
Comment (tyler-ball) [inline www/site/content/docs/reference/config.md]: Lets remove this line

File: www/site/content/docs/reference/config.md [hunk 5]
```diff
@@ -229,68 +193,61 @@ Used by
...
-Control the behavior of automatic update checking for Chef Workstation.
+These options are intended for development and troubleshooting of Chef Workstation tools. Their usage is not supported and is subject to change.
 
 #### Example
 
-```toml
-[updates]
-enable=true
-channel="current"
+```tom
```
Comment (tyler-ball) [inline www/site/content/docs/reference/config.md]: `tom` -> `toml` (saw at least 1 other instance too)

File: www/site/content/docs/reference/config.md (hunk 3 above)
Comment (afiune) [reply www/site/content/docs/reference/config.md]: @kagarmoe I have updated this doc, could you give it another look please? 😄

**Comment by afiune** [comment]
@tyler-ball @susanev @kagarmoe I have updated this PR, could y'all give it another review, please?

tenor-36414149

File: www/site/content/docs/reference/config.md [hunk 6]
```diff
@@ -379,3 +379,26 @@ Default
...
+
+#### Example
+
+```toml
+[features]
+analyze = true
+```
+
+Description
+: List of experimental features that can be turn on and off. In the above example, there is one turned on feature, the `analyze` feature. Every feature flag has a way to be turned on through the command line via environment variables. The respective environment variable for the `analize` feature would be `CHEF_FEAT_ANALYZE`. When set to any value, the feature will be turned on.
```
Comment (kagarmoe) [inline www/site/content/docs/reference/config.md]:
```suggestion
: List of experimental features.  Features are neither initiated nor enabled by default. Turn on a  feature by adding it to the `[features]` .  Set the feature to `true`  to enable and `false` to disable. Features added to the list must be enabled or disabled. The above example initiates and enables one feature, which is the `analyze` feature. You can also turn on and enable or disable any feature flag from the command line using an environment variable.  For example, `CHEF_FEAT_ANALYZE`  is the environment variable for the `analyze` feature and setting `CHEF_FEAT_ANALYZE=true` enables the `analyze` feature.
```

Comment (kagarmoe) [reply www/site/content/docs/reference/config.md]: I'm not sure that I've caught the nuances correctly. Let me know.

Comment (afiune) [reply www/site/content/docs/reference/config.md]: Looking now 👁]👁

File: www/site/content/docs/reference/config.md [hunk 7]
```diff
@@ -379,3 +379,26 @@ Default
...
+
+Values
+: `name = true`, `name = false`
+
+`CHEF_FEAT_[NAME]`
+: When set to any value, the feature will be turned on.
```
Comment (afiune) [inline www/site/content/docs/reference/config.md]: let us remove this too, thanks @kagarmoe

**Pull request merged**